src/main/java/uk/gov/hmcts/darts/common/repository/CourthouseRepository.java

```diff
@@ -34,4 +34,6 @@ AND securityRole.id IN (:roleIds)
 
     boolean existsByDisplayNameIgnoreCaseAndIdNot(String name, Integer id);
 
+    List<CourthouseEntity> findByIdIn(List<Integer> courthouseIds);
```

Comment:
Don't think we need this if we already have `findAllById` from ListCrudRepository?

But as per earlier comment, the existing view entity could either be simplified to just be plain old Integer columns - no ManyToOne joins needed if you are going to use a findAllById here or changing the query select and return type 
`uk.gov.hmcts.darts.common.repository.UserRolesCourthousesRepository#findAllCourthouseIdsByUserAccount`